src/tensors/abstracttensor.jl

```diff
@@ -3,53 +3,53 @@
 # Abstract Tensor type
 #----------------------
 """
-    abstract type AbstractTensorMap{S<:IndexSpace, N₁, N₂} end
+    abstract type AbstractTensorMap{E<:Number, S<:IndexSpace, N₁, N₂} end
 
-Abstract supertype of all tensor maps, i.e. linear maps between tensor products
-of vector spaces of type `S<:IndexSpace`. An `AbstractTensorMap` maps from
-an input space of type `ProductSpace{S, N₂}` to an output space of type
-`ProductSpace{S, N₁}`.
+Abstract supertype of all tensor maps, i.e. linear maps between tensor products of vector
+spaces of type `S<:IndexSpace`, with element type `E`. An `AbstractTensorMap` maps from an
```

Comment:
I would be up for consistently referring to `E` as the scalar type. I see why you chose `E` for element type, but I guess there is no conflict with calling it `T` as in the `AbstractArray{T,N}` convention?

Reply:
I think I did not go for `T` as that might be confusing with `TensorMap`, but I am more than happy to consistently change it back to `T`.